test/junit/scala/collection/IteratorTest.scala

```diff
@@ -192,4 +192,26 @@ class IteratorTest {
     assertSameElements(exp, res)
     assertEquals(8, counter) // was 14
   }
+  
+  // SI-9766
+  @Test def exhaustedConcatIteratorConcat: Unit = {
+    var i: Iterator[Int] = Iterator.empty
```

Comment:
This is the ticket example, but mutables are as hard to read in unit tests as in real code.